### PR TITLE
[continuous-integration] use lcov to generate coverage data

### LINKED ISSUE #5628: [codecov] code coverage threshold check is failing
PRs experiencing this issue:

- https://github.com/openthread/openthread/pull/5627
- https://github.com/openthread/openthread/pull/5624
- https://github.com/openthread/openthread/pull/5623
- https://github.com/openthread/openthread/pull/5614

### PR DESCRIPTION
**Pull request by Moandor-y**
This PR makes it use `lcov` to generate the coverage report and upload using the bash script, which may be more stable.

Resolves #5628.

### CODE REVIEW

**Comment by Moandor-y** [comment]
Codecov still shows wrong coverage data. The `ProcessDns` in CLI is called in expect tests but it shows uncovered: <https://codecov.io/gh/openthread/openthread/src/2d2fc92b2a82f0c17b67313a39c4ab70c34127cb/src/cli/cli.cpp>.
Looks like the Codecov itself is unstable.

File: tests/scripts/thread-cert/node.py [hunk 1]
```diff
@@ -160,10 +163,14 @@ def _shutdown_docker(self):
             if COVERAGE or OTBR_COVERAGE:
                 self.bash('service otbr-agent stop')
 
-                codecov_cmd = 'bash /tmp/codecov.bash -Z'
                 # Upload OTBR code coverage if OTBR_COVERAGE=1, otherwise OpenThread code coverage.
-                if not OTBR_COVERAGE:
-                    codecov_cmd += ' -R third_party/openthread/repo'
+                if OTBR_COVERAGE:
+                    codecov_cmd = 'bash /tmp/codecov.bash -Z'
```
Comment (simonlingoogle) [inline tests/scripts/thread-cert/node.py:168]: need to fix for OTBR too?

File: tests/scripts/thread-cert/node.py [hunk 2]
```diff
@@ -108,6 +109,8 @@ def _launch_docker(self):
             f'{self._rcp_device}:/dev/ttyUSB0',
             '-v',
             '/tmp/codecov.bash:/tmp/codecov.bash',
```
Comment (simonlingoogle) [inline tests/scripts/thread-cert/node.py:111]: Is codecov.bash still useful?

File: .github/workflows/posix.yml [hunk 3]
```diff
@@ -241,15 +265,43 @@ jobs:
...
+      run: |
+        sudo apt-get --no-install-recommends install -y lcov
+    - uses: actions/download-artifact@v2
       with:
-        fail_ci_if_error: true
+        path: coverage/
+    - name: Upload Coverage
+      run: |
+        ls -R coverage/
+        lcov $(for i in $(find coverage/ -type f -name '*.info'); do echo -a $i; done) -o final.info
```
Comment (simonlingoogle) [inline .github/workflows/posix.yml:306]: The command is complex and used many times. Can we encapsulate the logic in `script/test` using a new command?

**Review by simonlingoogle — COMMENTED** [review]
Should we adjust `after_n_builds` in codecov.yml too in this PR?

File: tests/scripts/thread-cert/node.py (hunk 1 above)
Comment (Moandor-y) [reply tests/scripts/thread-cert/node.py:168]: It seems OTBR is running the same tests too? If so I think it's good to change it to upload in the same way.

File: .github/workflows/posix.yml (hunk 3 above)
Comment (Moandor-y) [reply .github/workflows/posix.yml]: Done.

File: tests/scripts/thread-cert/node.py (hunk 1 above)
Comment (Moandor-y) [reply tests/scripts/thread-cert/node.py]: Done.

File: tests/scripts/thread-cert/node.py (hunk 2 above)
Comment (Moandor-y) [reply tests/scripts/thread-cert/node.py]: Done.

**Review by wgtdkp — APPROVED** [review]
LGTM 👍

**Review by bukepo — APPROVED** [review]
LGTM:+1:

**Review by simonlingoogle — COMMENTED** [review]
Great work! LGTM :+1: 

with nits

**Review by simonlingoogle — APPROVED** [review]

File: script/test [hunk 4]
```diff
@@ -272,7 +272,7 @@ do_build_otbr_docker()
     local otdir
     local otbrdir
     local otbr_options="-DOT_DUA=ON -DOT_MLR=ON -DOT_COVERAGE=ON"
-    local otbr_commit=${OTBR_COMMIT:-ce1cc3f9141571505bc9a8b705d2eb775849c92c} # ot-br-posix #532
+    local otbr_commit=${OTBR_COMMIT:-4ba806a0fbb4146f97baf7f8c2bb91ef7619cd3b} # ot-br-posix #532
```
[the remark this answers is not in the document]
Comment (Moandor-y) [reply script/test]: Reverted back to the "default master and specified by `OTBR_COMMIT`" method as the PR has been merged.

**Review by jwhui — APPROVED** [review]

**Pull request merged**